### ISSUE TITLE
Crash (on OSX) when choosing plugin to report

### ISSUE THREAD

**Issue by AsgerPetersen** [lutraconsulting/qgis-report-plugin#29]
1) Install report plugin
2) Activate it using the toolbar button
3) In the report dialog choose a plugin which has a GitHub tracker in the dropdown
Crash!

This is printed in the python console when I then choose "No" to report the crash:

```
Error in sys.excepthook:
Traceback (most recent call last):
  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 171, in qgis_excepthook
    showException(type, value, tb, None, messagebar=True)
  File "/Users/asger/.qgis2/python/plugins/report/report_plugin.py", line 55, in show_report_exception
    show_report_dialog(last_error)
  File "/Users/asger/.qgis2/python/plugins/report/report_plugin.py", line 36, in show_report_dialog
    report_dialog = main_widget.MainWidget(last_error)
  File "/Users/asger/.qgis2/python/plugins/report/main_widget.py", line 48, in __init__
    self._load_last_error()
  File "/Users/asger/.qgis2/python/plugins/report/main_widget.py", line 79, in _load_last_error
    self._find_plugin_from_exception(error)
  File "/Users/asger/.qgis2/python/plugins/report/main_widget.py", line 57, in _find_plugin_from_exception
    self._set_chosen_plugin(item)
  File "/Users/asger/.qgis2/python/plugins/report/main_widget.py", line 102, in _set_chosen_plugin
    self._plugin_selected()
  File "/Users/asger/.qgis2/python/plugins/report/main_widget.py", line 196, in _plugin_selected
    if self.selected_provider.is_valid():
  File "/Users/asger/.qgis2/python/plugins/report/providers/provider_base.py", line 29, in is_valid
    self._get_labels()
  File "/Users/asger/.qgis2/python/plugins/report/providers/github.py", line 82, in _get_labels
    labels = self._get("labels")
  File "/Users/asger/.qgis2/python/plugins/report/providers/github.py", line 67, in _get
    r = requests.get(url, headers=headers)
  File "/Users/asger/.qgis2/python/plugins/report/deps/requests-2.10.0-py2.py3-none-any.whl/requests/api.py", line 71, in get
  File "/Users/asger/.qgis2/python/plugins/report/deps/requests-2.10.0-py2.py3-none-any.whl/requests/api.py", line 57, in request
  File "/Users/asger/.qgis2/python/plugins/report/deps/requests-2.10.0-py2.py3-none-any.whl/requests/sessions.py", line 475, in request
  File "/Users/asger/.qgis2/python/plugins/report/deps/requests-2.10.0-py2.py3-none-any.whl/requests/sessions.py", line 585, in send
  File "/Users/asger/.qgis2/python/plugins/report/deps/requests-2.10.0-py2.py3-none-any.whl/requests/adapters.py", line 477, in send
requests.exceptions.SSLError: [Errno 20] Not a directory

Original exception was:
Traceback (most recent call last):
  File "/Users/asger/.qgis2/python/plugins/firstaid/debugwidget.py", line 193, in current_frame_changed
    self.go_to_frame(row)
  File "/Users/asger/.qgis2/python/plugins/firstaid/debugwidget.py", line 200, in go_to_frame
    self.source.openFile(filename)
  File "/Users/asger/.qgis2/python/plugins/firstaid/sourceview.py", line 65, in openFile
    self.setText(open(filename).read())
IOError: [Errno 20] Not a directory: '/Users/asger/.qgis2/python/plugins/report/deps/requests-2.10.0-py2.py3-none-any.whl/requests/adapters.py'
```

This happens on QGIS 2.14.0 on OSX. It doesn't happen on 2.14.2 on Win10.

**Comment by PeterPetrik** [comment]
@AsgerPetersen Can you please test if your SSH  connection to GitHub works from command line? https://help.github.com/articles/testing-your-ssh-connection/#platform-mac

**Comment by AsgerPetersen** [comment]
I get this error https://help.github.com/articles/error-permission-denied-publickey/#platform-mac

I am not sure I understand why. I literally use GitHub all the time.

**Comment by PeterPetrik** [comment]
That is why plugin raised this Exception. Problem is in the SSH key ... , plugin is only not catching the Exceptions triggered. We can fix that in the next version (I mean catching this Exception and reporting to the user that SSH is not correctly set up). I have a feeling that on Mac/Linux, user would need to set up SSH keys even before token (which would be probably the case for clients that do not usually use git).

**Comment by AsgerPetersen** [comment]
Yes, my setup fails this step: https://help.github.com/articles/error-permission-denied-publickey/#make-sure-you-have-a-key-that-is-being-used

It seems I have never used SSH to GitHub. All other embedded GitHub tools I have use personal access tokens over GitHub API instead.

**Comment by PeterPetrik** [comment]
With the fix of SSH keys, is the plugin working?

**Comment by AsgerPetersen** [comment]
No. Unfortunately. Now SSH from the terminal works:

```
$ ssh -T git@github.com
Hi AsgerPetersen! You've successfully authenticated, but GitHub does not provide shell access.
```

But the plugin still crashes with the same stack trace

**Comment by PeterPetrik** [comment]
Have you restarted QGIS after fix of SSH?

**Comment by AsgerPetersen** [comment]
Yes, I have restarted.

**Comment by AsgerPetersen** [comment]
@PeterPetrik ping me, if you need more testing.

**Comment by annalenahock** [comment]
I am having the same error... is there a solution so far?

**Comment by PeterPetrik** [comment]
@annalenahock I hope I can look into this tomorrow.

**Comment by PeterPetrik** [comment]
Problem was with SSL trusted certificates by OSX. Fixed by packaging https://github.com/certifi cacert.pem file with the plugin in case certifi is not installed on OS.

@annalenahock  @AsgerPetersen Fixed in version 0.2.1. Please update your plugin from QGIS plugin repository and let me know if the problem persists. Thanks.

**Comment by annalenahock** [comment]
Its working just fine!
